src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
I'm not 100% sure, but maybe you are on to something here.
The tests say "should parse ISO-based date strings as a valid min date value", but they specifically use Date objects. The commit that introduced that - https://github.com/angular/angular.js/commit/088545c1856ce1c3ec3416965dff65077a6e0523 - also said it closes #6755, but it looks to me that it should say Closes #6754, which is precisely about using Min/Max with Date objects: https://github.com/angular/angular.js/issues/6754
For #6755, there's another PR: https://github.com/angular/angular.js/commit/b3502835039178296b730b7526e5666b66ba9156

So I assume the intention was to allow JSONified Date objects; however this change was never documented.
I'd say that we should make it work and document it. I like your comparson approach: `new Date(iso).toString() === iso`

I assumed `Date.parse()` returns a Date object -_-. Yes, we probably want new Date().